### ISSUE TITLE
Expire old test runs

### ISSUE THREAD

**Issue by tomchipchase** [Shopify/ci-queue#131]
Looking at our graphs of redis memory usage, it looks like this gem only removes keys when it hits the memory limit of redis.

This is causing us some problems, as the redis SaaS we are using has a total memory limit for all our instances, and our db for the ci-test-queue using all the memory it can use means other databases have less memory available than they need to operate properly.

I thought I would investigate whether there was a way to clear out old test runs from the ci-queue. 
If this isn't currently possible, I'd be happy to put in the work to make it possible. My thought would be to add an optional ttl to the redis commands storing data, but I'd be interested in hearing alternative approaches.

We're using the rubygem version 0.16.0.

**Comment by casperisfine** [comment]
> it looks like this gem only removes keys when it hits the memory limit of redis.

Yeah, it's a lack of documentation. `ci-queue` currently expects the Redis server to have an eviction policy set to `allkeys-lru`.

So it's not the gem that clears the redis, it relies on expiration.

> I thought I would investigate whether there was a way to clear out old test runs from the ci-queue.

In theory we could clear the Redis state when reporting a successful run, however:

  - You can't clear failed run, because the whole point is that workers are idempotent. If you retry a worker it re-use the existing state.
  - It would be quite brittle.

> My thought would be to add an optional ttl to the redis commands storing data

That's something I'd be positive about. Beside the lacking documentation, requiring users to configure their Redis with `maxmemory-policy allkeys-lru` makes the gem harder to use.
